bundler/lib/bundler/compact_index_client/updater.rb

```diff
@@ -92,11 +92,8 @@ def slice_body(body, range)
 
       def checksum_for_file(path)
         return nil unless path.file?
-        # This must use File.read instead of Digest.file().hexdigest
-        # because we need to preserve \n line endings on windows when calculating
-        # the checksum
         SharedHelpers.filesystem_access(path, :read) do
-          SharedHelpers.digest(:MD5).hexdigest(File.read(path))
+          SharedHelpers.digest(:MD5).file(path).hexdigest
```

Comment:
I changed this method because one spec was failing with an invalid encoding error. I think this code was now incorrect since we started writing the versions file with "binary" encoding at https://github.com/rubygems/rubygems/pull/4362.

Reply:
Well, now we have some mostl likely related specs failing on Windows. I'll dig deeper into this and possibly move this fix/changes to a separate PR.

Reply:
Final version doesn't need these changes.